### PR TITLE
Standardize formatting for nextflow files

### LINKED ISSUE #480: Standardize formatting for nextflow files
Right now there is a bit of a mix of styles in the nextflow files in this repository. Most prominently, we have a mix of two-space and four-space tabs, so we should standardize on one or the other (I prefer 2 here, despite using 4 in Python, I guess because whitespace in nextflow isn't syntactically meaningful). A quick pass through the nextflow files could get this done in a single PR: We should be able to review it with `ignore whitespace` to verify that there are no code changes in that PR.

Looking forward a bit more, we should write up a bit of a style guide/editor setup instructions. I think this would include the use of `black` for Python and `styler`/tidyverse style for R scripts.

We might also include some examples of the way we want to handle `.map{}` functions and the like, `tuple()` vs. `[]`, etc.
These later changes might be more incremental in implementation, but it might be nice to have some guidelines written down. 

I tried to look for existing guidelines in the [nf-core](https://github.com/nf-core/) repositories but largely came up short (their [code formatting](https://nf-co.re/docs/contributing/code_formatting) section is mostly focused on `prettier` and `black`, neither of which actually format nextflow or groovy code). There are obviously conventions but not "guidelines" that I could find.

### PR DESCRIPTION
**Pull request by jashapiro**
Closes #480

I wend through all the nextflow files and made them all 2 space indented, then ran `black` and `styler` on all other files for good measure.

I also updated the `CONTRIBUTING.md` file to add a few notes related to formatting & the little note about `def` that we wanted as a reminder.

The _vast_ majority of changes here should be whitespace only, but some changes will still show up as text modifications, as there are changes made by formatters and me to break up some function definitions and calls to multiple lines in different ways than had been done before. 

There is also one code modification where I removed an unused variable from a function that a linter caught.

Other than that, there should be no functional changes to code, but please flag any that you might notice in review!

### CODE REVIEW

**Review by sjspielman — APPROVED** [review]
Quite the overhaul 🚀 !!

Overall things look fine to me. I left some comments for places you could tweak, totally at your discretion, and like 1-3 very small other comments. 

I don't feel that I need to see this again, but that said, some caveats about my review:

- I didn't broadly look at files to see if there are other spots which could be re-formatted. I just looked at `Files changed`.
- I hid all whitespace changes during review, under the assumption that the whitespace changes are correct.

If you feel this PR would benefit from me re-reviewing in ways different from ^, please feel free to dismiss my review and tell me to come back!

File: bin/predict_cellassign.py [hunk 1]
```diff
@@ -5,14 +5,15 @@
 # CellAssign is run and the predictions matrix is returned
 
 
+import argparse
 import os
+import re
+
```
[the remark this answers is not in the document]
Comment (jashapiro) [reply bin/predict_cellassign.py:11]: Yes. The convention is to separate base packages from ones that need to be installed.

File: integrate.nf [hunk 2]
```diff
@@ -118,11 +118,13 @@ process integration_report {
...
-                                  output_file = '${integration_report}', \
-                                  params = list(integration_group = '${integration_group}', \
-                                                integrated_sce = '${integrated_sce_file}', \
-                                                batch_column = 'library_id'))"
+    Rscript -e "rmarkdown::render( \
+      '${report_template}', \
+      output_file = '${integration_report}', \
+      params = list(integration_group = '${integration_group}', \
+                    integrated_sce = '${integrated_sce_file}', \
+                    batch_column = 'library_id') \
```
[the remark this answers is not in the document]
Comment (jashapiro) [reply integrate.nf:126]: Yeah, I had this at one point, but it was getting pretty ugly with all the backslashes, so I left it this way.

File: modules/af-features.nf [hunk 3]
```diff
@@ -147,31 +151,34 @@ workflow map_quant_feature{
...
       .map{it.drop(1)} // remove the first element (feature_barcode_file)
 
     // // if the rad directory has been created and repeat_mapping is set to false
     // create tuple of metdata map (read from output) and rad_directory to be used directly as input to alevin-fry quantification
     feature_rad_ch = feature_ch.has_rad
-      .map{meta -> tuple(Utils.readMeta(file("${meta.feature_rad_dir}/scpca-meta.json")),
-                         file(meta.feature_rad_dir, type: 'dir')
-                         )}
+      .map{meta -> tuple(
+        Utils.readMeta(file("${meta.feature_rad_dir}/scpca-meta.json")),
```
[the remark this answers is not in the document]
Comment (jashapiro) [reply modules/af-features.nf:179]: It is 2! (lined up under the a in `.map`)

File: bin/predict_cellassign.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sjspielman) [reply bin/predict_cellassign.py:11]: that was my guess!

File: modules/spaceranger.nf [hunk 4]
```diff
@@ -108,50 +108,53 @@ def getCRsamples(files_dir){
...
+      )}
+
+    // run spaceranger
+    spaceranger(spaceranger_reads)
+
+    // gather spaceranger output for completed libraries
+    // make a tuple of metadata (read from prior output) and prior results directory
+    spaceranger_quants_ch = spatial_channel.has_spatial
+      .map{meta -> tuple(
+        Utils.readMeta(file("${meta.spaceranger_results_dir}/scpca-meta.json")),
```
[the remark this answers is not in the document]
Comment (jashapiro) [reply modules/spaceranger.nf:148]: I thought about it, but I think that makes the tuple elements harder to read, so I'm leaving it for now

File: modules/cellsnp.nf [hunk 5]
```diff
@@ -78,15 +78,20 @@ workflow cellsnp_vireo {
...
+
+    // make a channel with: [meta, star_bam, star_bai, star_quant, meta_mpileup, vcf_file]
     star_mpileup_ch = starsolo_bam_ch.map{[it[0].library_id] + it} // add library id at start
-      .join(starsolo_quant_ch.map{[it[0].library_id] + it}, by: 0, failOnDuplicate: true, failOnMismatch: true) // join starsolo outs by library_id
+      .join(starsolo_quant_ch.map{[it[0].library_id] + it}, // join starsolo outs by library_id
+            by: 0, failOnDuplicate: true, failOnMismatch: true)
       .map{[it[0], it[1], it[2], it[3], it[5]]} // remove redundant meta
-      .join(mpileup_ch, by: 0, failOnDuplicate: true, failOnMismatch: true) // join starsolo and mpileup by library id
+      .join(mpileup_ch, // join starsolo and mpileup by library id
+            by: 0, failOnDuplicate: true, failOnMismatch: true)
```
[the remark this answers is not in the document]
Comment (jashapiro) [reply modules/cellsnp.nf]: done

**Pull request merged**